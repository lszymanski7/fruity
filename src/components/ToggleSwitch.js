import React from 'react'
import PropTypes from 'prop-types'

const ToggleSwitch = ({ colors, handleOnChange, icons, id, isChecked }) => {
    return (
        <div className="toggle-switch">
            <input
                checked={isChecked}
                className="toggle-switch__input"
                id={id}
                onChange={handleOnChange}
                type="checkbox"
            />
            <label
                className={
                    'toggle-switch__label' +
                    (colors ? (isChecked ? ' ' + colors[0] : ' ' + colors[1]) : '')
                }
                htmlFor={id}
            >
                {icons && (
                    <>
                        <span className={'toggle-switch__icon ' + icons[0]} />
                        <span className={'toggle-switch__icon ' + icons[1]} />
                    </>
                )}
                <span className="toggle-switch__slider" />
            </label>
        </div>
    )
}

ToggleSwitch.propTypes = {
    colors: PropTypes.array,
    handleOnChange: PropTypes.func.isRequired,
    icons: PropTypes.array,
    id: PropTypes.string.isRequired,
    isChecked: PropTypes.bool.isRequired
}

export { ToggleSwitch as default }
