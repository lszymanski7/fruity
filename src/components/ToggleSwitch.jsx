import React from 'react'
import PropTypes from 'prop-types'

const ToggleSwitch = ({ handleOnChange, isChecked }) => {
  return (
    <div
      className="toggle-switch"
      data-testid="theme-switch"
    >
      <input
        checked={isChecked}
        className="toggle-switch__input"
        id="theme-switch"
        onChange={handleOnChange}
        type="checkbox"
      />
      <label
        className="toggle-switch__label"
        htmlFor="theme-switch"
      >
        <span
          className="toggle-switch__icon moon-icon"
          data-testid="moon"
        />
        <span
          className="toggle-switch__icon sun-icon"
          data-testid="sun"
        />
        <span className="toggle-switch__slider" />
      </label>
    </div>
  )
}

ToggleSwitch.propTypes = {
  handleOnChange: PropTypes.func.isRequired,
  isChecked: PropTypes.bool.isRequired
}

export default ToggleSwitch
