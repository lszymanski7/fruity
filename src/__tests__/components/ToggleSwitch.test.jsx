import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import ToggleSwitch from '../../components/ToggleSwitch'

const props = {
  handleOnChange: jest.fn(),
  isChecked: false
}

describe('ToggleSwitch', () => {
  it('should be rendered correctly.', () => {
    const { container } = render(<ToggleSwitch {...props} />)
    expect(container).toMatchSnapshot()
  })

  it('should handle clicking the checkbox.', async () => {
    render(<ToggleSwitch {...props} />)
    const checkbox = screen.getByRole('checkbox')
    await userEvent.click(checkbox)
    expect(props.handleOnChange).toHaveBeenCalled()
  })

  it('should have a moon icon.', () => {
    render(<ToggleSwitch {...props} />)
    const icon = screen.getByTestId('moon')
    expect(icon).toHaveClass('moon-icon')
  })

  it('should have a sun icon.', () => {
    render(<ToggleSwitch {...props} />)
    const icon = screen.getByTestId('sun')
    expect(icon).toHaveClass('sun-icon')
  })
})
