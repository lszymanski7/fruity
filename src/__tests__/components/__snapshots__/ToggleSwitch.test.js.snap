// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch should be rendered correctly. 1`] = `
<div>
  <div
    class="toggle-switch"
    data-testid="theme-switch"
  >
    <input
      class="toggle-switch__input"
      id="theme-switch"
      type="checkbox"
    />
    <label
      class="toggle-switch__label"
      for="theme-switch"
    >
      <span
        class="toggle-switch__icon moon-icon"
        data-testid="moon"
      />
      <span
        class="toggle-switch__icon sun-icon"
        data-testid="sun"
      />
      <span
        class="toggle-switch__slider"
      />
    </label>
  </div>
</div>
`;
